onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -42,6 +53,33 @@ class OpenDataViewSet(ETagsMixin, CacheControlMixin,
     MAX_INSTANCES_PER_REQUEST = 1000
     pagination_class = StandardPageNumberPagination
 
+    def clean_data_fields(self, data, xform):  # pylint disable=R0201
+        """
+        Streamlines the row header fields
+        with the column header fields for the same form.
+        """
+        if len(data) > 1:
+            result = []
+            for row in data:
+                # Get the column headers
+                # using the xform id string
+                headers = xform.get_headers()
+                tableau_headers = replace_special_characters_with_underscores(
+                                    headers)
+
+                # Include fields from the column headers
+                # that are not present in the row headers
+                column_headers = set(tableau_headers)
+                row_headers = set(row)
+                diff = column_headers.difference(row_headers)
+
+                # set default values for these additional fields
+                left_out_fields = {
+                    field: "" for field in diff if not field.startswith('_')}
+                row.update(left_out_fields)
+                result.append(row)
+        return result
```

Comment:
```suggestion
        result = []
        if data:
           headers = xform.get_headers()
           tableau_headers = remove_meta_fields(headers)
            for row in data:
                # Include fields from the column headers
                diff = set(tableau_headers).difference(set(row))

                # Shouldn't the headers here be cleared of all `_` fields ?
                # Other than '_id' ?
                left_out_fields = {
                    field: "" for field in diff if not field.startswith('_')}
                row.update(left_out_fields)
                result.append(row)
        return result
```

Reply:
Is it possible to have this functionality within the Data serializer ? I'm guessing that's where it's used ?

You can do so by using the [`exclude` variable](https://www.django-rest-framework.org/api-guide/serializers/#specifying-which-fields-to-include)